File: 1046-max-consecutive-ones-iii/1046-max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        left = longest = zeros = 0
        for right in range(len(nums)):
            if(nums[right] == 0): zeros += 1
            if zeros > k:
                if(nums[left] == 0): zeros -= 1
                left += 1
            longest = max(longest, right - left + 1)
        return longest

File: 1046-max-consecutive-ones-iii/1046-max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int zeros = 0, longest = 0;
        int left = 0, n = nums.size();
        for(int right = 0; right < n; right++){
            if(nums[right] == 0) zeros++;
            if(zeros > k){
                if(nums[left] == 0) zeros--;
                left++;
            }
            longest = max(longest, right - left + 1);
        }
        return longest;
    }
};